crates/uv-build/src/error.rs

```diff
@@ -99,51 +104,96 @@ pub enum Error {
 enum MissingLibrary {
     Header(String),
     Linker(String),
-    PythonPackage(String),
+    BuildDependency(String),
+    DeprecatedModule(String, Version),
 }
 
 #[derive(Debug, Error)]
 pub struct MissingHeaderCause {
     missing_library: MissingLibrary,
+    package_name: Option<PackageName>,
+    package_version: Option<Version>,
     version_id: String,
 }
 
 impl Display for MissingHeaderCause {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
         match &self.missing_library {
             MissingLibrary::Header(header) => {
-                write!(
-                    f,
-                    "This error likely indicates that you need to install a library that provides \"{}\" for {}",
-                    header, self.version_id
-                )
+                if let (Some(package_name), Some(package_version)) =
+                    (&self.package_name, &self.package_version)
+                {
+                    write!(
+                        f,
+                        "This error likely indicates that you need to install a library that provides \"{header}\" for {package_name}@{package_version}",
```

Comment:
I love this type of specific error handling btw (I really miss the Cargo style great error messages in Python in general). 

For context, over half of Python app build failures on our platform fail during the `pip install` step (not surprising given it's the biggest point of user-provided input), so improving UX there is something I'm very interested in. 

We've tried a small amount of log parsing based suggestions in the past, but (a) hadn't had a chance to put more effort into them, (b) the package manager really has more context than something wrapping it to try and make more intelligent suggestions - so having uv handle these instead of the buildpack would be ideal.

I presume you would be interested to know more about any real world failure modes that are confusing for users, for cases where uv doesn't already handle them well? (For now all such examples will be based on pip, but I'm sure most will translate.)

Reply:
Yes, absolutely. Any data or info you can share on common errors would be much appreciated.